### PR TITLE
Clarify `Texture Atlas rework` migration guide

### LINKED ISSUE #1025: Migration Guide 0.13 - Texture Atlas rework migration unclear
Looking at the [sprite_sheet](https://github.com/bevyengine/bevy/blob/release-0.12.1/examples/2d/sprite_sheet.rs) example if you were using `TextureAtlasSprite` in a query it's currently not immediately obvious how to migrate that from the [migration guide](https://bevyengine.org/learn/migration-guides/0-12-to-0-13/#texture-atlas-rework).

One person (bless them) ran into this so there are probably more people out there running into it. For more context:
https://discord.com/channels/691052431525675048/1209175742081142836/1209175742081142836

Additionally you have `atlas_handle` in the code shown in the guide, but it's unclear what it is or where it came from.

Since I haven't worked with Bevy 2D stuff yet I'm a bit hesitant to make a PR myself.

### PR DESCRIPTION
**Pull request by TrialDragon**
Clarifies the `Texture Atlas rework` migration guide. Uses @rparrett's [migration guide notes](https://gist.github.com/rparrett/508035fa19d3324db0adf8e509e7affd#httpsgithubcombevyenginebevypull10099) and https://github.com/bevyengine/bevy/pull/5103 's change log as a base. 

Resolves #1025

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

**Review by rparrett — CHANGES_REQUESTED** [review]
I can't make a specific suggestion right now, but the first paragraph has some typos / grammatical errors and could use a closer look.

Just some general feedback:

Overall in my notes I tried to lay it out like

> `Thing` was removed and had `A` and `B`
> - Where `A` can now be found
> - Where `B` can now be found

Totally fine with rewriting those bullet points as a paragraph, but a lot of the language in the current text in this PR is pretty redundant which makes the changes hard to follow for me.

**Review by rparrett — APPROVED** [review]

File: content/learn/migration-guides/0.12-to-0.13.md [hunk 1]
```diff
@@ -884,6 +884,14 @@ The `TextureAtlas` asset that previously contained both the atlas layout and ima
 
 `SpriteSheetBundle` now uses a `Sprite` instead of a `TextureAtlasSprite` component and a `TextureAtlas` component instead of a `Handle<TextureAtlaslayout>`.
 
+`DynamicTextureAtlasBuilder::add_texture` takes an addition `&Handle<Image>` parameter.
```
Comment (rparrett) [inline content/learn/migration-guides/0.12-to-0.13.md]:
```suggestion
`DynamicTextureAtlasBuilder::add_texture` takes an additional `&Handle<Image>` parameter.
```

File: content/learn/migration-guides/0.12-to-0.13.md [hunk 2]
```diff
@@ -884,6 +884,14 @@ The `TextureAtlas` asset that previously contained both the atlas layout and ima
...
+
+`TextureAtlasLayout::from_grid` no longer takes a `Handle<Image>` parameter.
+
+`TextureAtlasBuilder::finish` now returns a `Result<(TextureAtlasLayout, Handle<Image>), _>`.
+
+`UiAtlasImage` was removed. The atlas bundle is now identical to `ImageBundle` with an additional `TextureAtlas`.
```
Comment (rparrett) [inline content/learn/migration-guides/0.12-to-0.13.md:893]:
```suggestion
`UiAtlasImage` was removed. Use `ImageBundle` and add an additional `TextureAtlas` component.
```

File: content/learn/migration-guides/0.12-to-0.13.md [hunk 3]
```diff
@@ -884,6 +884,14 @@ The `TextureAtlas` asset that previously contained both the atlas layout and ima
 
 `SpriteSheetBundle` now uses a `Sprite` instead of a `TextureAtlasSprite` component and a `TextureAtlas` component instead of a `Handle<TextureAtlaslayout>`.
 
+`DynamicTextureAtlasBuilder::add_texture` takes an addition `&Handle<Image>` parameter.
+
+`TextureAtlasLayout::from_grid` no longer takes a `Handle<Image>` parameter.
+
+`TextureAtlasBuilder::finish` now returns a `Result<(TextureAtlasLayout, Handle<Image>), _>`.
```
Comment (rparrett) [inline content/learn/migration-guides/0.12-to-0.13.md]: This doesn't seem right

File: content/learn/migration-guides/0.12-to-0.13.md (hunk 2 above)
Comment (TrialDragon) [reply content/learn/migration-guides/0.12-to-0.13.md:893]: This is confusing. https://docs.rs/bevy/latest/bevy/prelude/struct.AtlasImageBundle.html does exist; why shouldn't this be used like the changelog suggests?

Comment (rparrett) [reply content/learn/migration-guides/0.12-to-0.13.md:893]: Oops, this is nonsense from me.

Comment (rparrett) [reply content/learn/migration-guides/0.12-to-0.13.md]: I think that it was `UiTextureAtlasImage` that was removed though.

**Comment by TrialDragon** [comment]
Okay, should be good to go now. This technically mentions changes from later PRs since they changed the same API (then poorly documented it in their migration guides such that it probably wouldn't be seen).
